File: java/src/main/java/mx/iteso/adapter/word/IWordDocument.java
package mx.iteso.adapter.word;

import mx.iteso.adapter.Format;
import mx.iteso.adapter.Image;

/**
 * Word Doc.
 */
public interface IWordDocument {
     /**
      *
      * @return format.
      */
     Format getFormat();

     /**
      *
      * @return background.
      */
     Image getBackground();

     /**
      *
      * @param msOfficeVersion .
      */
     void setMSOfficeVersion(float msOfficeVersion);

     /**
      *
      * @return license.
      */
     MSLicense getLicense();

     /**
      *
      * @return is invalid falseee.
      */
     boolean restrictEditIfLicenseIsInvalid();

}

File: python/src/main/python/IWordDocument.py
from abc import ABC

class IWordDocument(ABC):
    def getFormat(self):
        pass

    def getBackground(self):
        pass

    def setMSOfficeVersion(self, msOfficeVersion):
        pass

    def getLicense(self):
        pass

    def restrictEditIfLicenseIsInvalid(self, msLicense):
        pass